taichi/transforms/simplify.cpp

```diff
@@ -661,9 +661,6 @@ class BasicBlockSimplify : public IRVisitor {
   }
 
   void visit(LinearizeStmt *stmt) override {
-    if (is_done(stmt))
-      return;
-
     if (stmt->inputs.size() && stmt->inputs.back()->is<IntegerOffsetStmt>()) {
```

Comment:
Let's comment out this clause. I need to think a bit more about if this part is necessary.

Reply:
I think `stmt` no longer exists once `set_done(stmt);` and the following code are executed. Added the comment back anyway.

Reply:
Sounds good!